pkg/topology/builder.go

```diff
@@ -218,9 +218,11 @@ func (b *Builder) evaluateTrafficSplit(topology *Topology, trafficSplit *split.T
 			setTrafficSplitWithErr(topology, trafficSplit, svcKey, err)
 			b.Logger.Errorf("Error building topology for TrafficSplit %q: %v", tsKey, err)
 
-			return
+			continue
 		}
 
+		var err error
+
 		// As required by the SMI specification, backends must expose at least the same ports as the Service on
 		// which the TrafficSplit is.
 		for _, svcPort := range svc.Ports {
```

Comment:
It's a bit hard to understand these loops. Maybe splitting these loops will make the code easier to understand?

Reply:
We refactor this in #550. Did we want to wait for that PR to merge and then update with this fix?

Reply:
You're right, I think it's simpler to merge #550 and then update this fix.